### PR TITLE
Views refactor

### PR DESCRIPTION
**Pull request by dinohamzic**

### CODE REVIEW

File: app/views/spree/shared/_head.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<%== meta_data_tags %>
+<%= canonical_tag(current_store.url) %>
+<%= csrf_meta_tags %>
+
+<%= favicon_link_tag 'favicon.ico' %>
+
+<%= stylesheet_link_tag 'spree/frontend/all', media: 'screen' %>
+<%= javascript_include_tag 'spree/frontend/all' %>
+
```
Comment (aleph1ow) [inline app/views/spree/shared/_head.html.erb]: here we removed retro compatibility
```
<!--[if lt IE 9]>
  <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.6/html5shiv.min.js"></script>
<![endif]-->
```

https://github.com/solidusio/solidus/blob/master/frontend/app/views/spree/shared/_head.html.erb

File: app/views/spree/shared/_products.html.erb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  paginated_products ||= products
+%>
+
+<% content_for :head do %>
+  <% if paginated_products.respond_to?(:total_pages) %>
+    <%= rel_next_prev_link_tags paginated_products %>
+  <% end %>
+<% end %>
+
+<% if params.key?(:keywords) %>
```
Comment (aleph1ow) [inline app/views/spree/shared/_products.html.erb]: Here we intentionally removed the no_products_found message?
https://github.com/solidusio/solidus/blob/master/frontend/app/views/spree/shared/_products.html.erb#L13

File: app/views/spree/checkout/existing_payment/_gateway.html.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<tr id="<%= dom_id(wallet_payment_source, 'spree')%>">
+  <td><%= wallet_payment_source.payment_source.name %></td>
+  <td><%= wallet_payment_source.payment_source.display_number %></td>
+  <td><%= wallet_payment_source.payment_source.month %></td>
+  <td><%= wallet_payment_source.payment_source.year %></td>
+  <td>
+    <%= radio_button_tag "order[wallet_payment_source_id]", wallet_payment_source.id, default, class: "existing-cc-radio" %>
```
[the remark this answers is not in the document]
Comment (dinohamzic) [reply app/views/spree/checkout/existing_payment/_gateway.html.erb]: The class is related to some JavaScript behavior that we can refactor later. I would leave it for reference for now.

File: app/views/spree/home/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<% content_for :sidebar do %>
+  <%= render partial: 'spree/shared/taxonomies' %>
+<% end %>
+
+<article data-hook="homepage_products">
```
[the remark this answers is not in the document]
Comment (dinohamzic) [reply app/views/spree/home/index.html.erb]: Removed 👍

File: app/views/spree/layouts/spree_application.html.erb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<!doctype html>
```
[the remark this answers is not in the document]
Comment (dinohamzic) [reply app/views/spree/layouts/spree_application.html.erb]: Removed 👍

File: app/views/spree/orders/_line_item.html.erb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<% variant = line_item.variant -%>
+
+<%= order_form.fields_for :line_items, line_item do |item_form| -%>
```
[the remark this answers is not in the document]
Comment (dinohamzic) [reply app/views/spree/orders/_line_item.html.erb]: Removed 👍

File: app/views/spree/orders/_line_item.html.erb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        <p><%= t('spree.out_of_stock') %></p>
+      <% end %>
+      <p><%= line_item_description_text(line_item.description) %></p>
+    </td>
+    <td><%= line_item.single_money.to_html %></td>
+    <td><%= item_form.number_field :quantity, min: 0, class: "line_item_quantity", size: 5 %></td>
+    <td><%= line_item.display_amount.to_html unless line_item.quantity.nil? %></td>
+    <td><%= link_to image_tag('icons/delete.png'), '#', class: 'delete', id: "delete_#{dom_id(line_item)}" %></td>
+  </tr>
+<% end -%>
```
[the remark this answers is not in the document]
Comment (dinohamzic) [reply app/views/spree/orders/_line_item.html.erb]: Removed 👍

File: app/views/spree/products/_properties.html.erb [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    <%= t('spree.properties')%>
+  </h2>
+  <table>
+    <tbody>
+      <% @product_properties.each do |product_property| %>
+        <tr>
+          <td><%= product_property.property.presentation %></td>
+          <td><%= product_property.value %></td>
+        </tr>
+      <% end %>
```
[the remark this answers is not in the document]
Comment (dinohamzic) [reply app/views/spree/products/_properties.html.erb]: Nice catch, added.

File: app/views/spree/shared/_filters.html.erb [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+              params[:search][filter[:scope]] &&
+              params[:search][filter[:scope]].include?(val.to_s) ? "checked" : ""
+          %>
+          <li>
+            <input
+              type="checkbox"
+              id="<%= label %>"
+              name="search[<%= filter[:scope].to_s %>][]"
+              value="<%= val %>"
+              <%= checked %>
```
[the remark this answers is not in the document]
Comment (dinohamzic) [reply app/views/spree/shared/_filters.html.erb]: The same logic is on line 16. I just moved it in a variable for clarity.

File: app/views/spree/shared/_head.html.erb (hunk 1 above)
Comment (dinohamzic) [reply app/views/spree/shared/_head.html.erb]: Retro compatibility is something we should not worry about at the moment. Maybe we should also discuss internally what level of support we want to have for this starter.

File: app/views/spree/shared/_products.html.erb (hunk 2 above)
Comment (dinohamzic) [reply app/views/spree/shared/_products.html.erb]: Yes, it was an error in the logic of the old frontend, a part of the code was never called.

File: app/views/spree/checkout/edit.html.erb [hunk 10]
```diff
@@ -0,0 +1,27 @@
+<article>
```
[the remark this answers is not in the document]
Comment (dinohamzic) [reply app/views/spree/checkout/edit.html.erb]: Nice catch, fixed.

**Pull request merged**